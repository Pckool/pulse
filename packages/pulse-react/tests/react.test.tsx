import * as React from 'react';
import Pulse, { State, usePulse } from '../lib';
import * as renderer from 'react-test-renderer';

let //
  App: Pulse,
  StringState: State<string>,
  BooleanState: State<boolean>, // TODO: use this somewhere, anywhere!
  TestComponent: React.FC<{}>;

const TestString = 'This is a string! Wow, so glorious!';

beforeAll(() => {
  App = new Pulse({ noCore: true });
});

describe('usePulse() | Typings', () => {
  beforeEach(() => {
    StringState = App.State(TestString);
    BooleanState = App.State(false);
  });

  test('State typings', () => {
    TestComponent = () => {
      const [myStringDestructured] = usePulse([StringState]);
      const myStringSingle = usePulse(StringState);
      const myHookArray: [string] = usePulse([StringState]);

      expect(typeof myStringDestructured).toBe('string');
      expect(myStringDestructured.length > 1).toBeTruthy();

      expect(typeof myStringSingle).toBe('string');
      expect(myStringSingle.length > 1).toBeTruthy();

      expect(myHookArray).toHaveLength(1);
      expect(typeof myHookArray).toBe('object');
      expect(typeof myHookArray[0]).toBe('string');

      return (
        <div>
          <p>{myStringSingle}</p>
          <p>{myStringDestructured}</p>
          <p>{myHookArray[0]}</p>
        </div>
      );
    };

    const testRender = renderer.create(<TestComponent />);
    let tree = testRender.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
