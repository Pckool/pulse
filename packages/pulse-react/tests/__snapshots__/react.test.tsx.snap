// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`usePulse() | Typings State typings 1`] = `
<div>
  <p>
    This is a string! Wow, so glorious!
  </p>
  <p>
    This is a string! Wow, so glorious!
  </p>
  <p>
    This is a string! Wow, so glorious!
  </p>
</div>
`;
